package ftp_rpc;

import "ftp_model.proto";

message Message
{
    optional int64 id = 1;
    optional uint32 req_sec = 2;
    optional uint32 req_usec = 3;

    optional DetailMessage detail_message = 4;
}

message DetailMessage
{
    oneof choice
    {
        ftp_model.DownloadReq download_req = 1;
        ftp_model.DownloadResp download_resp = 2;

        ftp_model.UploadReq upload_req = 3;
        ftp_model.UploadResp upload_resp = 4;
    }
}
